src/ray/raylet/node_manager.cc

```diff
@@ -517,16 +580,25 @@ void NodeManager::ProcessClientMessage(
       worker_pool_.RegisterWorker(std::move(worker));
       DispatchTasks();
     } else {
-      // Register the new driver.
-      JobID job_id = from_flatbuf(*message->driver_id());
-      worker->AssignTaskId(job_id);
+      // Register the new driver. Note that here the driver_id in RegisterClientRequest
+      // message is actually the ID of the driver task, while client_id represents the
+      // real driver ID, which can associate all the tasks/actors for a given driver,
+      // which is set to the worker ID.
+      JobID driver_task_id = from_flatbuf(*message->driver_id());
+      worker->AssignTaskId(driver_task_id);
+      worker->AssignDriverId(from_flatbuf(*message->client_id()));
       worker_pool_.RegisterDriver(std::move(worker));
-      local_queues_.AddDriverTaskId(job_id);
+      local_queues_.AddDriverTaskId(driver_task_id);
```

Comment:
not related to this PR, but I think it's time to split `ProcessClientMessage` into a few smaller functions. I can do that later.